src/backend/node.rs

```diff
@@ -280,6 +282,41 @@ fn get_total_account_balance(
     Some(account_data.free + account_data.reserved + account_data.frozen)
 }
 
+// defined here: https://github.com/subspace/subspace/blob/5d8b65740ff054b01ebcbaf5a905e74274c1a5d0/crates/subspace-core-primitives/src/pieces.rs#L803
+const PIECE_SIZE: usize = 1048672/* the actual piece size from your runtime */;
+// TODO: needs to be moved to runtime constants or somewhere else
+const SLOT_PROBABILITY: (u64, u64) = (1, 6);
+
+#[allow(dead_code)]
+pub(crate) fn get_total_space_pledged<Block, Client>(
+    client: &Client,
+    block_hash: H256,
+) -> Result<u128, ApiError>
+where
+    Block: BlockT<Hash = H256> + sp_api::__private::BlockT,
+    Client: ProvideRuntimeApi<Block>,
+    Client::Api: SubspaceApi<Block, FarmerPublicKey>,
+{
+    let current_solution_range = client
+        .runtime_api()
+        .solution_ranges(block_hash)
+        .map(|solution_ranges| solution_ranges.current)
+        .expect("Failed to get current solution range");
```

Comment:
This is not really a proof. Check other usages of `.expect()`.

Reply:
Not sure, if you meant this.
I refactored to:
```diff
-let current_solution_range = client
-        .runtime_api()
-        .solution_ranges(block_hash)
-        .map(|solution_ranges| solution_ranges.current)?;
+ let current_solution_range = client
+        .runtime_api()
+        .solution_ranges(block_hash)
+        .map_err(|_| "Failed to get solution ranges")?
+        .current;

Reply:
This is what I meant: https://github.com/subspace/subspace/blob/main/CONTRIBUTING.md#code-style

Error handling is fine, though your example above hides the actual error, so it will be hard to debug when/if it happens.

Reply:
After reading it, what I got to know is...we should avoid panicking as much as possible.
Can this code be considered:
```rust
let current_solution_range = match runtime_api
        .solution_ranges(block_hash)
        .map(|solution_range| solution_range.current)
    {
        Ok(res) => res,
        Err(_) => {
            return Err(ApiError::UnknownBlock(
                "Failed to get solution ranges at this block".to_string(),
            ))
        }
    };

let slot_probability = runtime_api
    .chain_constants(block_hash)
    .map(|chain_constants| chain_constants.slot_probability())
    .unwrap_or((1, 6));

let max_pieces_in_sector = runtime_api
    .max_pieces_in_sector(block_hash)
    .unwrap_or(1000_u16);
 ```
 
 assuming to set a default value of `slot_probability`, `max_pieces_in_sector`.